frontend/public/components/utils/list-dropdown.jsx

```diff
@@ -0,0 +1,146 @@
+    const {loaded, loadError} = nextProps;
+
+    if (loadError) {
+      this.setState({
+        title: <div className="cos-error-title">Error Loading {nextProps.desc}</div>,
+      });
+      return;
+    }
+
+    if (!loaded) {
+      return;
+    }
+    const state = {};
+    const { resources, dataFilter, selectedKey } = nextProps;
+
+    state.items = {};
+    _.each(resources, ({data}, kindLabel) => {
+      _.reduce(data, (acc, resource) => {
+        if (!dataFilter || dataFilter(resource)) {
+          acc[`${resource.metadata.name}-${kindLabel}`] = {kindLabel, name: resource.metadata.name};
+        }
+        return acc;
+      }, state.items);
+    });
+
+    // did we switch from !loaded -> loaded ?
+    if (!this.props.loaded || !selectedKey) {
+      state.title = <span className="text-muted">{nextProps.placeholder}</span>;
+    }
+
+    if (selectedKey) {
+      const item = state.items[selectedKey];
+      // item may not exist if selectedKey is a role and then user switches to creating a ClusterRoleBinding
+      if (item) {
+        state.title = <ResourceName kind={item.kindLabel} name={item.name} />;
+      }
+    }
+
+    this.setState(state);
+  }
+
+  shouldComponentUpdate (nextProps, nextState) {
+    if (_.isEqual(this.state, nextState)) {
+      return false;
+    }
+    return true;
+  }
+
+  render () {
+    const {desc, fixed, placeholder, id, loaded} = this.props;
+    const items = {};
+    const sortedItems = _.keys(this.state.items).sort();
+
+    _.each(this.state.items, (v, key) => items[key] = <ResourceName kind={v.kindLabel} name={v.name} />);
+
+    const {selectedKey} = this.state;
+
+    const Component = fixed
+      ? items[selectedKey]
+      : <Dropdown
+        autocompleteFilter={this.autocompleteFilter}
+        autocompletePlaceholder={placeholder}
+        items={items}
+        sortedItemKeys={sortedItems}
+        selectedKey={selectedKey}
+        title={this.state.title}
+        onChange={this.onChange}
+        id={id}
+        menuClassName="dropdown-menu--text-wrap" />;
+
+    return <div>
+      { Component }
+      { loaded && _.isEmpty(items) && <p className="alert alert-info"><span className="pficon pficon-info" aria-hidden="true"></span>No {desc} found or defined.</p> }
+    </div>;
+  }
+}
+
+export const ListDropdown = props => {
+  const resources = _.map(props.resources, resource => _.assign({ isList: true, prop: resource.kind }, resource));
+  return <Firehose resources={resources}>
+    <ListDropdown_ {...props} />
+  </Firehose>;
+};
+
+ListDropdown.propTypes = {
+  dataFilter: PropTypes.func,
+  desc: PropTypes.string,
+  // specify both key/kind
+  selectedKey: PropTypes.string,
+  selectedKeyKind: PropTypes.string,
+  fixed: PropTypes.bool,
+  resources: PropTypes.arrayOf(PropTypes.shape({
+    kind: PropTypes.string.isRequired,
+    namespace: PropTypes.string,
+  })).isRequired,
+  placeholder: PropTypes.string,
+};
+
+const NsDropdown_ = props => {
```

Comment:
Did you want to include the NSDropdown in this component?  I ask because there are several other components in the binding code that also build from the ListDropDown.

Reply:
Yeah, NsDropdown is used a bunch now, so I pulled it into the list-dropdown module, too.

We should look at component naming, but wanted to move the code first.